packages/core/src/components/ImageViewer/index.tsx

```diff
@@ -0,0 +1,167 @@
+  static defaultProps = {
+    height: 'none',
+    rotation: ROTATION,
+    scale: SCALE,
+    width: 'none',
+  };
+
+  state = {
+    dragging: false,
+    imageLocation: { x: 0, y: 0 },
+    lastMouseLocation: { x: 0, y: 0 },
+  };
+
+  componentDidMount() {
+    document.addEventListener('mousemove', this.handleMouseMove, false);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('mousemove', this.handleMouseMove, false);
+  }
+
+  handleMouseDown = (event: MouseEvent | React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+    event.preventDefault();
+    this.setState({
+      dragging: true,
+      lastMouseLocation: {
+        x: event.pageX,
+        y: event.pageY,
+      },
+    });
+  };
+
+  handleMouseUp = (event: MouseEvent | React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+    event.preventDefault();
+    this.setState({
+      dragging: false,
+      lastMouseLocation: { x: 0, y: 0 },
+    });
+  };
+
+  handleMouseMove = (event: MouseEvent) => {
+    event.preventDefault();
+    this.setState(({ dragging, imageLocation, lastMouseLocation }) => {
+      if (dragging) {
+        const xDiff = lastMouseLocation.x - event.pageX;
+        const yDiff = lastMouseLocation.y - event.pageY;
+
+        return {
+          imageLocation: {
+            x: imageLocation.x - xDiff,
+            y: imageLocation.y - yDiff,
+          },
+          lastMouseLocation: {
+            x: event.pageX,
+            y: event.pageY,
+          },
+        };
+      }
+
+      return null;
+    });
+  };
+
+  getTransformStyle() {
+    const {
+      imageLocation: { x, y },
+    } = this.state;
+    const { scale = SCALE, rotation = ROTATION } = this.props;
+
+    const radian = (rotation / 180) * Math.PI;
+    const sinRotation = Math.sin(radian);
+    const cosRotation = Math.cos(radian);
+    const translateX = (y * sinRotation + x * cosRotation) / scale;
+    const translateY = (y * cosRotation - x * sinRotation) / scale;
+
+    return {
+      transform: `scale(${scale}) rotate(${rotation}deg) translateY(${translateY}px) translateX(${translateX}px)`,
+    };
+  }
+
+  render() {
+    const { alt, borderless, height, src, width, cx, styles } = this.props;
+
+    return (
+      <div
+        className={cx(styles.container, borderless && styles.container_borderless)}
+        role="presentation"
+        onMouseDown={this.handleMouseDown}
+        onMouseUp={this.handleMouseUp}
+        style={{ width, height }}
+      >
+        <div className={cx(styles.image)} style={this.getTransformStyle()}>
+          <ResponsiveImage
+            contain
+            noShadow
+            alt={alt}
+            borderRadius={0}
+            maxWidth={width}
+            maxHeight={height}
+            src={src}
+          />
+        </div>
+      </div>
+    );
+  }
+}
+
+export { ZoomControls, RotateControls };
+
+export default withStyles(({ ui }) => ({
+  container: {
+    border: `1px solid ${ui.border}`,
```

Comment:
my bad, this whole thing would be `ui.border`: `border: ui.border`

Reply:
no problem, I was admittedly rushing